scripts/common/proxy.sh

```diff
@@ -47,6 +47,56 @@ function proxy_bootstrap() {
     fi
 }
 
+# check_proxy_config tries to check if is possible connect with the registry
+# If the result is an error that is  NOT UNAUTHORIZED such as:
+# {"errors":[{"code":"UNAUTHORIZED","message":"authentication required","detail":null}]}
+# Then, it means that the proxy configuration does not affected the connection and it is not invalid.
+function check_proxy_config() {
+    if [ -n "$CONTAINERD_VERSION" ]; then
+        logStep "Checking registry config with Containerd"
+
+        local config_file="/etc/containerd/config.toml"
+
+        # Check if the config file exists
+        if [ ! -f "$config_file" ]; then
+            logWarn "File $config_file not found. Unable to check connection"
+            return
+        fi
+
+        # Get IP and ca_file path from the config.toml file
+        local ip=$(cat "$config_file" | grep -oP '(?<=plugins\."io.containerd.grpc.v1.cri"\.registry\.configs\.").*(?="\.tls\])')
+        local ca_file=$(cat "$config_file" | grep -oP '(?<=ca_file = ").*(?=")')
+
+        # Echo containerd Proxy config:
+        local proxy_config_file="/etc/systemd/system/containerd.service.d/http-proxy.conf"
+        if [ -f "$proxy_config_file" ]; then
+           echo "Containerd HTTP proxy config:"
+           grep -v -e '^\[Service\]' -e '^# Generated by kURL' "$proxy_config_file"
+           echo ""
+        fi
+
+        # Execute curl command and capture the output
+        echo "curl --cacert $ca_file https://$ip/v2/"
+        local response=""
+        if output=$(curl --cacert "$ca_file" "https://$ip/v2/" 2>&1); then
```

Comment:
it is very unclear to me the intention of this change and how this is testing containerd

Reply:
@emosbaugh,

Our goal is validate if is possible to connect to registry, so we are here mainly:

Getting the registry configured on the containerd (/etc/containerd/config.toml)

```
[plugins."io.containerd.grpc.v1.cri".registry.configs."10.96.2.223".tls]
  ca_file = "/etc/kubernetes/pki/ca.crt"
```

Then, we are call the registry API: 

```
$ curl --cacert /etc/kubernetes/pki/ca.crt https://10.96.2.223/v2/
{"errors":[{"code":"UNAUTHORIZED","message":"authentication required","detail":null}]}
```

If the proxy be invalid for example or we have any issue to connect to the registry we would not face UNAUTHORIZED or `{}` (success when is possible authenticate)

Have you a better idea how can we verify the scenario?

Reply:
But what is the expectation here? Is it supposed to be using the proxy? Curl is not configured the same as containerd. This will behave differently depending on the environment.

Reply:
Closing this one as sorted out since we agreed to change it out.